service_container/tags.rst

```diff
@@ -564,3 +564,216 @@ application handlers.
+            ->addArgument(new TaggedIteratorArgument('app.handler', 'key'));
+
+After compilation the ``HandlerCollection`` service is able to iterate over your
+application handlers. But we want to retrieve a specific service by it's
+``key`` attribute. As it's an iterator, we can use ``iterator_to_array``
+to get an array and then retrieve the ``handler_two`` handler :
+
+.. code-block:: php
+
+    // src/Handler/HandlerCollection.php
+    namespace App\Handler;
+
+    class HandlerCollection
+    {
+        public function __construct(iterable $handlers)
+        {
+            $handlers = iterator_to_array($handlers);
+
+            $handler_two = $handlers['handler_two']:
+            $handler_one = $handlers['handler_one'];
+        }
+    }
+
+
+.. tip::
+
+    You can omit the ``index_attribute_name`` attribute, by implementing a static
+    method named like this : ``getDefaultIndexAttributeNameName``
+
+    Based on the previous example ``App\Handler\One`` should look like :
+
+    .. code-block:: php
+
+        // src/Handler/One.php
+        namespace App\Handler;
+
+        class One
+        {
+            public static function getDefaultIndexName() {
+                return 'handler_one';
+            }
+        }
+
+    And the configuration :
+
+    .. configuration-block::
+
+        .. code-block:: yaml
+
+            # config/services.yaml
+            services:
+                App\Handler\One:
+                    tags:
+                        - { name: 'app.handler', priority: 20 }
+
+                ....
+
+        .. code-block:: xml
+
+            <!-- config/services.xml -->
+            <?xml version="1.0" encoding="UTF-8" ?>
+            <container xmlns="http://symfony.com/schema/dic/services"
+                xmlns:xsi="http://www.w3.org/2001/XMLSchema-instance"
+                xsi:schemaLocation="http://symfony.com/schema/dic/services
+                    http://symfony.com/schema/dic/services/services-1.0.xsd">
+
+                <services>
+                    <service id="App\Handler\One">
+                        <tag name="app.handler" priority="20" />
+                    </service>
+
+                    ...
+                </services>
+            </container>
+
+        .. code-block:: php
+
+            // config/services.php
+            $container->register(App\Handler\One::class)
+                ->addTag('app.handler', ['priority' => 20]);
+
+            ...
+
+    You also can define the name of the static method to implement on each service
+    with the ``default_index_method`` attribute on the argument.
+
+    Based on the previous example ``App\Handler\One`` should look like :
+
+    .. code-block:: php
+
+        // src/Handler/One.php
+        namespace App\Handler;
+
+        class One
+        {
+            public static function someFunctionName()
+            {
+                return 'handler_one';
+            }
+        }
+
+    And the configuration :
```

Comment:
```suggestion
    And the configuration:
```

@weaverryan not sure how we handle the colon in the symfony docs, with or without leading whitespace?